circleci.go

```diff
@@ -396,6 +397,36 @@ func (c *Client) BuildOpts(account, repo, branch string, opts map[string]interfa
 	return build, nil
 }
 
+// BuildByProject triggers a build by project (this is the only way to trigger a build for project using Circle 2.1)
+func (c *Client) BuildByProject(vcsType VcsType, account string, repo string, branch string, revision string, tag string) error {
```

Comment:
I thought about allowing the caller to pass in a `buildParams` map instead, but I decided to follow the existing pattern of treating `branch` as a top-level arg, and treated `revision` and `tag` similarly. This makes it easier to enforce the constraints of the circle API (if `tag` is included, `branch` and `revision` must not be present). Happy to accept a map instead though if you'd prefer!

Reply:
:+1: I think it makes sense to structure the input, but I'm not crazy about inviting library users to accidentally provide conflicting arguments (tag with either branch or revision).

What would you think of having two methods: `BuildByProjectByTag` and `BuildByProjectByBranch`? I think this would be inline with the existing function signatures. I'd like to generally revisit the signatures as part of switching to the `v1.1` API which requires the vcs type in the URL as that will require backwards incompatible changes anyway.